docs/DeveloperGuide.md

```diff
@@ -1,38 +1,536 @@
+with little ambiguity. Command validation can then be performed sequentially, facilitating
+efficient checks.
+
+### Implementation
+
+The main function of `Parser` is to decode user input and extract relevant information to be
+used by other classes. Hence, it supports 3 main methods as listed below.
+1. `parseCommand`
+2. `getContentAfterKeyword`
+3. `extractQuestionDifficulty`
+
+`parseCommand`
+
+This method processes raw user input from CLI to extract information to be assembled into a
+`Command` object. It determines the type of command by reading the commandType from the
+very start of the user input.
+
+The following sequence diagram shows the implementation of `parseCommand`.
+
+![](./UML/Parser.jpg)
+
+`getContentAfterKeyword`
+
+This method acts as a helper method to assist in extracting the payload following an argument
+in the user input. It is used in the constructing of `Command` objects when command-specific
+information need to be extracted from arguments in the user input.
+
+It simply breaks up an entire ```/argument [payload]``` String into ```/argument``` and
+```[payload]``` and returns the `[payload]`. Hence, any String passed into this method must
+take on the form of ```/argument [payload]``` or an exception will be thrown and handled by
+printing an error message to the CLI.
+
+`extractQuestionDifficulty`
+
+This method acts as a helper method for extracting an `QnDifficulty` enum type from a user
+text input. This is used for setting question difficulty during the construction of a
+`Question` , or the execution of a `CommandMarkDifficulty`.
+
+If the string passed to this method does not describe an existing question difficulty level, an
+exception will be thrown and handled by printing an error message to the CLI.
+
+## Command Components
+
+### Help Command - Display Commands
+
+When executed, this command will execute a standard Final String containing all 
+the commands available for the user to use, as well as the format they are meant
+to be written in. 
+
+This same String is also displayed when an invalid command is used.
+
+### Short Command - Add Short Answer Question to the Quiz
+
+One of the supported question formats is the Short Answer Question, in which a question
+can be answered with a phrase of a few words, which will be matched to a corresponding answer
+of sufficient similarity.
+
+It is parsed as: `short [question]/[answer]/[module]/[difficulty]`
+1. `[question]` is the question, phrased as asking the user (i.e. What is 2 + 2? )
+2. `[answer]` is the answer or possible answer(s) that the user must
+   input to count as correct. It is case-insensitive. (i.e. 4, four)
+3. `[module]` is the module that the question belongs in (i.e. CS2113)
+4. `[difficulty]` is the difficulty of the question for sorting later (i.e. Hard)
+
+*Condensed Class Diagram - Does not contain all attributes & methods
+![](./UML/AddShortCommand.jpg)
```

Comment:
Should there be a link between Question and QuestionList?

Reply:
![image](https://github.com/nus-cs2113-AY2324S1/tp/assets/23655240/4c61206f-bcbe-48d6-9c34-53f642134b16)